File: vite-vue-ts-frontend/src/modules/time-trackings/types/dto.ts
import type { UserBaseResponse } from "../../users/types/dto";

export type AddRequest = {
  summary: string;
  spentTime: number;
};

export type UpdateRequest = {
  id: string;
  summary: string;
  spentTime: number;
};

export type TimeTrackingResponse = {
  id: string;
  createdBy: UserBaseResponse;
  createdAt: number;
  summary: string;
  spentTime: number;
};

export type SearchResponse = {
  timeTrackings: TimeTrackingResponse[];
};

File: golang-backend/internal/handlers/tasktimetrackinghandler/dto.go
package tasktimetrackinghandler

import "github.com/aportela/doneo/internal/handlers/userhandler"

type addRequest struct {
	Summary   string `json:"summary"`
	SpentTime uint64 `json:"spentTime"`
}

type updateRequest struct {
	ID        string `json:"id"`
	Summary   string `json:"summary"`
	SpentTime uint64 `json:"spentTime"`
}

type taskTimeTrakingResponse struct {
	ID        string                       `json:"id"`
	CreatedBy userhandler.UserBaseResponse `json:"createdBy"`
	CreatedAt int64                        `json:"createdAt"`
	Summary   string                       `json:"summary"`
	SpentTime uint64                       `json:"spentTime"`
}

type searchResponse struct {
	TimeTrackings []taskTimeTrakingResponse `json:"timeTrackings"`
}
